docs/adr/adr-0018-how-to-give-feedback-after-a-user-action.md

```diff
@@ -0,0 +1,239 @@
+---
+kind: '📌 Docs/Architecture Decision Records'
+---
+# ADR 0018: How to give feedback after a user action?
+
+🗓️ 2022-08-26 · ✍️ Pierre de Soyres
+
+This ADR tries to explain why and how we implemented a mechanism that triggers notifications after user actions.
+
+## Context?
+
+When a user performs an action (e.g. clicking on a button), it is common practice to give feedback after the action has been completed.
+When something goes wrong, showing an error message may help the user understand what happened and maybe provide insights on how to correct the error.
+Also, showing a message after an action succeeded can comfort the user that the action has been taken into account successfully.
+
+We started very simply by displaying only error messages near the button that triggered the action.
+
+## Problems?
+
+* The error messages were hard to place nicely and there was no consistency.
+* The component had to maintain different error states which complicates the code a lot.
+* We did not handle the case where the action succeeded.
+
+## What we want
+
+Before explaining the solution, here is what we want.
+
+* We want all messages to be located in the same area of the screen.
+* We want the messages to appear without disturbing the user workflow.
+* We want the messages to disappear automatically after a small amount of time (few seconds).
+* We want the messages to be decorated in such a way that the intention is easily understood (information, success, warning, error).
+
+The `toast` pattern seems to be what we need.
+
+Technically speaking, we want to allow developers to integrate the component in a totally decoupled manner so that it doesn't have any adherence to the technology or framework used (in our case [Lit](https://lit.dev/)).
+Finally, we try to be pragmatic and not provide too many features that we don't need right now.
+
+## What we implemented
+
+We created a system composed of two Web Components:
+
+* a `<cc-toast>` component which displays the message.
+* a `<cc-toaster>` component which is the container of the `<cc-toast>`.
```

Comment:
This sentence feels like it can only contain one.

Reply:
What do you think if I replace that by:
> a `<cc-toaster>` component which is the container of the `<cc-toast>` elements.